src/App.tsx

```diff
@@ -1,6 +1,86 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { FaAngleDoubleRight } from "react-icons/fa";
 function App() {
-  return <h1>Hello</h1>;
+  const url = "https://course-api.com/react-tabs-project";
+
+  const [data, setData] = useState<any[]>([]);
+  const [index, setIndex] = useState<number>(0);
+  const [error, setError] = useState<boolean>(false);
+
+  useEffect(() => {
+    try {
+      setError(false);
+      const fetching = async () => {
+        const response = await fetch(url);
+        const res = await response.json();
+        setData(res);
+        console.log(res);
+      };
+      fetching();
+    } catch (err) {
+      setError(true);
+    }
+  }, []);
+  if (error === true) {
+    return <h1 className="loading">Error</h1>;
+  }
+  if (data.length === 0) {
+    return <h1 className="loading">Loading...</h1>;
+  } else {
```

Comment:
instead of if..else.. block we can simply do
if() {
return(...)
}

return(..);